### PR TITLE
feature/eng-137-function-invoke

### PR DESCRIPTION
**Pull request by dmikey**
fixes build, and then invoke

### CODE REVIEW

File: src/commands/function/build.ts [hunk 1]
```diff
@@ -24,6 +24,10 @@ const createManifest = (
   return manifest;
 };
 
+const renameWasm = (path: string, oldName: string, newName: string) => {
+  execSync(`mv ${oldName} ${newName}`, { cwd: path, stdio: "inherit" });
+};
+
 export const run = (options: {
   debug: boolean;
   name: string;
```
Comment (sugardave) [inline src/commands/function/build.ts:33]: So, yeah....this renaming became obsolete when I made my (failing) stab at letting init handle it via modifying ascript.json, which should now be fixed by https://github.com/txlabs/blockless-cli/pull/30 so it can be dropped like a bad habit

File: src/commands/function/build.ts [hunk 2]
```diff
@@ -32,6 +36,7 @@ export const run = (options: {
 }) => {
   const { debug, name, path, rebuild } = options;
   // check for and store unmodified wasm file name to change later
+  const defaultWasm = debug ? "debug.wasm" : "release.wasm";
   const buildDir = `${path}/build`;
   const wasmName = `${name}${debug ? "-debug" : ""}.wasm`;
   const wasmArchive = `${name}.tar.gz`;
```
Comment (sugardave) [inline src/commands/function/build.ts:42]: ditto

File: src/commands/function/build.ts [hunk 3]
```diff
@@ -43,15 +48,17 @@ export const run = (options: {
       cwd: path,
       stdio: "inherit",
     });
+
+    if (existsSync(`${buildDir}/${defaultWasm}`)) {
+      renameWasm(buildDir, defaultWasm, wasmName);
+    }
   };
```
Comment (sugardave) [inline src/commands/function/build.ts:55]: Definitely axe this

File: src/commands/function/build.ts [hunk 4]
```diff
@@ -43,15 +48,17 @@ export const run = (options: {
...
 
   try {
-    if (!fs.existsSync(`${buildDir}/${wasmName}`)) build();
+    if (!fs.existsSync(`${buildDir}/${wasmName}`) || rebuild) build();
   } catch (err) {
     build();
   }
 
-  if (rebuild) build();
-
```
Comment (sugardave) [inline src/commands/function/build.ts:55]: lgtm!

File: src/commands/function/invoke.ts [hunk 5]
```diff
@@ -17,9 +18,17 @@ export const run = (options: any) => {
...
+      let manifestData = fs.readFileSync(`${path}/build/manifest.json`, "utf8");
+      let manifest = JSON.parse(manifestData);
+      manifest.entry = `${path}/build/${manifest.entry}`;
+      fs.writeFileSync(`${path}/build/manifest.json`, JSON.stringify(manifest));
+
+      // pass in stdin to the runtime
+      execSync(`echo "" | ${runtimePath} build/manifest.json`, {
         cwd: path,
         stdio: "inherit",
       });
```
Comment (sugardave) [inline src/commands/function/invoke.ts:34]: looks good

**Review by sugardave — COMMENTED** [review]
tl;dr
ditch the rename stuff, it is unnecessary once #30 is merged
